MSAL/src/native_auth/client_application_wrapper/silent_token/factory/MSALNativeAuthSilentTokenProviderFactory.swift

```diff
@@ -0,0 +1,30 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.  
+
+class MSALNativeAuthSilentTokenProviderFactory: MSALNativeAuthSilentTokenProviderBuildable {
```

Comment:
We need it to be able to pass the `MSALPublicClientApplicationConfig` to it, otherwise the `silentTokenProvider` would need to be created when the account gets created also and it would need to be injected

Reply:
Let me know if I'm missing something, but if you go to `MSALNativeAuthUserAccountResult+internal`, line 48, can you do:

```guard let silentTokenProvider = try? MSALNativeAuthSilentTokenProvider(configuration: config, challengeTypes: challengeTypes) else {...}```

instead of using the `TokenProviderFactory`?

Reply:
We need to inject the creator of the `silentTokenProvider` to be able to test it. Because the creation of the `silentTokenProvider` happens here (at line 48) we can't test this method unless we somehow inject the thing that creates the `silentTokenProvider`

Reply:
We can create the `silentTokenProvider` in `MSALNativeAuthResultFactory` and inject it to the `MSALNativeAuthUserAccountResult`. This still allows to test the code, and removes the factory class.

https://github.com/AzureAD/microsoft-authentication-library-for-objc/compare/spetrescu/update-idtoken-account...diegoje/update-idtoken-account-2?expand=1

Reply:
Indeed, that's pretty much what I had in mind originally, we would move the error handling of the creation to that class. I did indeed do a Subclass of the PublicClient rather than a Wrapper. @nilo-ms What's your thoughts on this?

Reply:
As discussed offline, we need to keep the factory in order to test that we are passing the correct values while creating the `MSALPublicClientApplicationConfig` needed for the `MSALNativeAuthSilentTokenProvider`.

Pending task is to create such tests for the factory.

Thank you @nilo-ms for your input.